libraries/Microsoft.Bot.Builder.LanguageGeneration/Evaluator.cs

```diff
@@ -270,7 +271,13 @@ public override object VisitNormalTemplateString([NotNull] LGFileParser.NormalTe
                 return result[0];
             }
 
-            return string.Join(string.Empty, result);
+            var resultStr = string.Join(string.Empty, result);
```

Comment:
Markdown transform should happens at the output stage of lg in case of some repeated work.
for example:
```
# template1
- ${template2()}

# template2
- text with \r\n
```

when evaluate `template1`, first we would get ` text with \r\n` -> ` text with \r\n\r\n`

and then back to template1, finally, we would get `text with \r\n\r\n\r\n\r\n`,
It is totally wrong.

Reply:
fixed